src/coreclr/tools/aot/ObjWriter/debugInfo/dwarf/dwarfGen.cpp

```diff
@@ -331,6 +367,7 @@ static void EmitVarLocation(MCObjectStreamer *Streamer,
       IsByRef = true;
     case ICorDebugInfo::VLT_STK2:
       IsStk2 = true;
+    case ICorDebugInfo::VLT_FPSTK:
```

Comment:
Did you actually hit this case? I would not expect it to be hit on. Floating point stack is a legacy 32-bit x86 that are not using anymore even on 32-bit x86.

Reply:
It is hit. I was tired of seeing assertion. This change was fishy to me too. But I do not know where to start looking. With that hint, I try to pin-point what can cause code to hit that location.

Reply:
I have looked around. `VLT_FPSTK` is actually used for regular local variables that do not live on floating point stack. It does not look right; but it may done that way for some legacy reasons. So your change is ok.